### ISSUE TITLE
$75 for help compiling Sparkle with nw.js (node-webkit)

### ISSUE THREAD

**Issue by btjones** [sparkle-project/Sparkle#550]
**TL;DR**
Help me edit a gypi file to add Sparkle to my mac app. I'll pay you $75 if I you help me get it compiling correctly.

---

I am trying to do a custom build of [nw.js](https://github.com/nwjs/nw.js/) (formerly named node-webkit) with Sparkle. I just need suggestions for things to try and if your suggestion gets it compiling, I'll send you $75 via Paypal.

I am building nwjs 0.12.2 on a mac and believe the issue is probably related to framework search paths in [nw.gypi](https://github.com/nwjs/nw.js/blob/master/nw.gypi).

You can see the changes I've tried to make to nw.gypi here: https://gist.github.com/btjones/dd85898f662ce6ab7d40/revisions

The changes in that gist DO compile BUT once I add `#import <Sparkle/Sparkle.h>` to any `*.mm` file (specifically [`src/browser/native_window_mac.mm`](https://github.com/nwjs/nw.js/blob/master/src/browser/native_window_mac.mm)) and build I get "fatal error: 'Sparkle/Sparkle.h' file not found". I've also tried lots of variations on the above changes to nw.gypi with no luck.

The one thing that is working properly is the copy task because when I remove the `#import` from `native_window_mac.mm` and build nwjs.app successfully, the framework is in the proper place (`nwjs.app/Contents/Frameworks/Sparkle.framework`).

Ideally I wouldn't need to put the framework in `/Library/Frameworks` and could instead use @loader_path (or whatever) so that I can store the framework in my project / repo and use a relative path.

So... What I need is to be able to compile with the framework in place when I add `#import <Sparkle/Sparkle.h>` to `src/browser/native_window_mac.mm` and the $75 is yours. I can take it from there. Thanks!

**Comment by kainjow** [comment]
Did you ever get this working? I had to integrate Sparkle with CMake projects at my work and in order for the headers to be found, I had to use the -F compiler flag with the path to the directory containing Sparkle.framework, so I imagine you'd have to do the same with however it's done with Gyp.

**Comment by btjones** [comment]
@kainjow I put it aside for a while so I have not gotten it working yet. I'll let you know if the -F compiler flag does the trick. Thanks!
